applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/explicit_central_differences_scheme.hpp

```diff
@@ -522,19 +548,22 @@ class ExplicitCentralDifferencesScheme
         array_1d<double, 3>& r_middle_angular_velocity = (itCurrentNode)->GetValue(MIDDLE_ANGULAR_VELOCITY);
         array_1d<double, 3>& r_current_angular_acceleration = (itCurrentNode)->FastGetSolutionStepValue(ANGULAR_ACCELERATION);
 
-        for (IndexType kk = 0; kk < DoF; ++kk) {
-        if (nodal_inertia[kk] > numerical_limit)
-            r_current_angular_acceleration[kk] = r_current_residual_moment[kk] / nodal_inertia[kk];
-        else
-            r_current_angular_acceleration[kk] = 0.0;
+        const IndexType initial_k = DoF == 3 ? 0 : 2;
```

Comment:
what are u doing here? at least put a comment to explain why this is needed

Reply:
It is because of the rotation dofs. In2D you only need ROTATION_Z, I will put a comment